SQF/dayz_code/compile/keyboard.sqf

```diff
@@ -2,331 +2,337 @@
+          uiSleep 0.5;
+        };
+      };
+    } else {
+      call dayz_autoRunOff;
     };
-	_dze_f = {
-		 if (!_ctrlState && !_altState) then {DZE_F = true;};
-	};
-	_dze_q = {
-		 if (!_ctrlState && !_altState) then {DZE_Q = true;};
-		 if (!_ctrlState && _altState) then {DZE_Q_alt = true;};
-		 if (_ctrlState && !_altState) then {DZE_Q_ctrl = true;};
-	};
-	_dze_z = {
-		 if (!_ctrlState && !_altState) then {DZE_Z = true;};
-		 if (!_ctrlState && _altState) then {DZE_Z_alt = true;};
-		 if (_ctrlState && !_altState) then {DZE_Z_ctrl = true;};
-	};
-	_autoRun = {
-		if (!dayz_autoRun) then {
-			dayz_autoRun = true;
-			dayz_autoRunThread = [] spawn {
-				_weapon = currentWeapon player;
-				while {dayz_autoRun} do {
-					// SurfaceIsWater does not work for ponds
-					// Check weapon to detect Arma action (dayz action is handled in dz_fn_switchWeapon)
-					if (player != vehicle player or (surfaceIsWater getPosASL player) or ((call fn_nearWaterHole) select 0) or (currentWeapon player != _weapon) or r_fracture_legs) exitWith {
-						call dayz_autoRunOff;
-					};
-					player playAction "FastF";
-					uiSleep 0.5;
-				};
-			};
-		} else {
-			call dayz_autoRunOff;
-		};
-		_handled = true;
-	};
-	_filterCheat = {
-		//Overriding default engine handling does not stop cheat input, need manual disableUserInput too
-		_handled = [displayNull,_dikCode,_shiftState] call dze_filterCheats;
-	};
-	_openGroups = {
-		if (dayz_requireRadio && !("ItemRadio" in items player)) then {
-			localize "STR_EPOCH_NEED_RADIO" call dayz_rollingMessages;
-		} else {
-			if (isNull findDisplay 80000) then {
-				if (!isNil "dayz_groupInit") then {[] spawn dayz_openGroupDialog;};
-			} else {
-				findDisplay 80000 closeDisplay 2;
-			};
-		};
-		_handled = true;
-	};
-	_muteSound = {
-		call player_toggleSoundMute;
-		_handled = true;
-	};
-    _rifle = {
-		2 call dz_fn_switchWeapon;
-        _handled = true;
+    _handled = true;
+  };
+  _filterCheat = {
+    //Overriding default engine handling does not stop cheat input, need manual disableUserInput too
+    _handled = [displayNull,_dikCode,_shiftState] call dze_filterCheats;
+  };
+  _openGroups = {
+    if (dayz_requireRadio && !("itemRadio" in items player)) then {
+      localize "STR_EPOCH_NEED_RADIO" call dayz_rollingMessages;
+    } else {
+      if (isNull findDisplay 80000) then {
+        if (!isNil "dayz_groupInit") then {[] spawn dayz_openGroupDialog;};
+      } else {
+        findDisplay 80000 closeDisplay 2;
+      };
     };
-    _pistol = {
-		3 call dz_fn_switchWeapon;
+    _handled = true;
+  };
+  _muteSound = {
+    call player_toggleSoundMute;
+    _handled = true;
+  };
+  _rifle = {
+    2 call dz_fn_switchWeapon;
+    _handled = true;
+  };
+  _pistol = {
+    3 call dz_fn_switchWeapon;
+    _handled = true;
+  };
+  _melee = {
+    //Second rifle if DZE_TwoPrimaries = 2 (dayz_onBack)
+    4 call dz_fn_switchWeapon;
+    _handled = true;
+  };
```

Comment:
DIK_4 which handles the throwable is not used by default so I commented the throwable code